strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/ParallelDownloadRemoteArtifactTest.java

```diff
@@ -56,8 +60,38 @@
     public void setup()
         throws Exception
     {
-        remoteArtifactInputStream = new RemoteArtifactInputStreamStub(jarArtifact);
-        prepareArtifactResolverContext(remoteArtifactInputStream, true);
+        prepareArtifactResolverContext(null, true);
```

Comment:
why this and `prepareArtifactResolverContext` was changed?

Reply:
`ParallelDownloadRemoteArtifactTest` overrides `RetryDownloadArtifactTestBase.prepareArtifactResolverContext` 

`ParallelDownloadRemoteArtifactTest#testConcurrentDownload` works concurrently over the same `InputStream` (first argument of original `prepareArtifactResolverContext`). There is a `try-with-resources Statement` in the `ProxyRepositoryArtifactResolver#fetchRemoteResource` (`try (InputStream is = new BufferedInputStream(new ProxyRepositoryInputStream(client, repositoryPath)))`. Threads were closing underlying  `InputStream` so some next Thread faced `Stream was closed` Exception. Overriden method replaces the same instance of mocked `InputStream` by new instance per each Thread.

Reply:
>  Threads were closing underlying InputStream so some next Thread faced Stream was closed Exception.

yes, and this was actually a goal of this test to check that artifact was fetched only once within multiple threads.

Reply:
so the behavior that you describe is correct within this test and assumed to be so.

Reply:
so we shouldn't change it :)

Reply:
In the concurrent live scenario the won't be the same `InputStream`. There are no exceptions within test.

Reply:
same `InputStream` let us to check the case that only one thread will get valid stream, and other threads should get artifact from hosted cache.

Reply:
We need a way to force fetch artifact from remote. Method mentioned above is the right one. I'll pay attention to your comments in the evening.

Reply:
like I already commented before you probably need another test to check force fetching within invalidation, this test have another purpose